### PR TITLE
Provision for showing components in header bar along with actions.

### PR DESCRIPTION
**Pull request by rmallah**
currently the actions only allow ActionIcons to be placed in the actionbar
 there can be use cases where a generic component may be useful. Eg
 a toggle switch as a quick access to some setting in app. This
 feature shall allow headerwidgets to be set to a list of arbitrary
 components in the same manner in which action items are set.

### CODE REVIEW

**Comment by rmallah** [comment]
Video of the Functionality:
https://drive.google.com/file/d/0B4yfAVMLmbQxMktxcXhMR041NlE/view?usp=sharing

**Comment by redgrapetech** [comment]
Hi @plfiorini  , 
kindly have a look at this pull request.
regds
mallah.

File: src/imports/controls/AppBar.qml [hunk 1]
```diff
@@ -150,6 +162,16 @@ ToolBar {
         spacing: 24 - 2 * leftButton.margin
 
         Repeater {
+                model: appBar.headerwidgets.length
+
+                delegate: Loader {
+                   id: widgetItem
```
Comment (plfiorini) [inline src/imports/controls/AppBar.qml]: Why a Loader instead of a container with child components?

**Comment by plfiorini** [comment]
I was planning to port some of the code from https://github.com/papyros/qml-material/blob/develop/src/window/ActionBar.qml which is much more complete.

Take a look at the extendedContentView and customContent

**Comment by redgrapetech** [comment]
@plfiorini  thanks that you had a look at it. 
I saw the referenced link. at the end i wanted to have that functionality
in fluid as i am using  it (fluid) in my app.  In case the functionality of 
extendedContentView and customContent can be used for the same 
function please feel free to port/incorporate the much for complete code
and ignore/dispose this pull request.

thanks & regards. Mallah.

**Comment by rmallah** [comment]
removing this request , as more generalized plans are in roadmap.

**Pull request closed without merging**